src/__tests__/organization.tests.ts

```diff
@@ -0,0 +1,319 @@
+      //Arrange
+      let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+      //Act
+      let payload = TdeiObjectFaker.getOrganization();
+      payload.address = '';
+
+      const oraganizationResponse = oraganizationApi.createOrganization(payload);
+
+      //Assert
+      await expect(oraganizationResponse).rejects.toMatchObject({response: { status: 400 } });
+    });
+
+    it('When creating new organization with invalid polygon, Expect to return HTTP Status 400', async () => {
+      //Arrange
+      let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+      //Act
+      let payload = TdeiObjectFaker.getOrganization();
+      payload.polygon = TdeiObjectFaker.getInvalidPolygon();
+
+      const oraganizationResponse = oraganizationApi.createOrganization(payload);
+
+      //Assert
+      await expect(oraganizationResponse).rejects.toMatchObject({ response: { status: 400 } });
+     });
+    });
+  });
+
+  describe("Update Organization", () => {
+    describe("Auth", () => {
+      it("When no auth token provided, Expect to return HTTP status 401", async () => {
+        //Arrange
+        let oraganizationApi = new OrganizationApi(configurationWithoutAuthHeader);
+        //Act
+        const request = async () => {
+          await oraganizationApi.updateOrganization(<Organization>{})
+        }
+        //Assert
+        expect(request()).rejects.toMatchObject({response: { status: 401 } })
+      });
+    });
+
+    describe("Functional", () => {
+      it("When updating new oraganization, Expect to return newly updated org id", async () => {
+        //Arrange
+        let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+        //Act
+        const organizationResponse = await oraganizationApi.updateOrganization(TdeiObjectFaker.getOrganization());
+        //Assert
+        expect(organizationResponse.status).toBe(200);
+        console.log(organizationResponse);
+        expect(organizationResponse.data).toBeGreaterThan(0);
+  
+      });
+      });
+
+      describe("Validation", () => {
+        it("When updating new oraganization with empty name, Expect to return HTTP Status 400", async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+          //Act
+          let payload = TdeiObjectFaker.getOrganization();
+          payload.org_name = '';
+    
+          const oraganizationResponse = oraganizationApi.updateOrganization(payload);
+    
+          //Assert
+          await expect(oraganizationResponse).rejects.toMatchObject({response: { status: 400 } });
+        });
+    
+        it("When updating new oraganization with empty phone, Expect to return HTTP Status 400", async () => {
+          //Arrange
+          let oraganizationApi = new OrganizationApi(configurationWithAuthHeader);
+          //Act
+          let payload = TdeiObjectFaker.getOrganization();
```

Comment:
Please look at the implementation in pathways for similar case

Reply:
fixed